### DISCUSSION TITLE
how to import tracing_subscriber?

### DISCUSSION THREAD

**Discussion by evbo** [aws/aws-lambda-rust-runtime#627] [General]
all the docs show this variable but do not show where it's imported from. Can it be clarified?

**Comment by evbo** [comment]
it appears to be a separate dependency:
https://github.com/awslabs/aws-lambda-rust-runtime/blob/main/examples/http-basic-lambda/Cargo.toml#L18

**Comment by greenwoodcm** [comment]
yeah it's referenced directly from the crate name.  does that clarify?

**Comment by timClicks** [comment]
@evbo, when you say "all the docs", which is the document that you encountered this first? It may be worth adding some context there.

**Reply by evbo** [reply]
I actually have realized today that "_" characters search very poorly in github, but if you search "subscriber" or "tracing" you can see in each readme example where it uses it without any explanation of what it's for or where it comes from.

I'm actually not totally sure how errors are supposed to be logged? Aside from println!, nothing seems to end up in cloudwatch.

Maybe this isn't a bad thing though, as I should be handling these errors in a state machine and properly messaging them...

**Reply by evbo** [reply]
@timClicks do I need to use tracing to get errors logged to cloudwatch? Or can I just set the log_level env variable? The docs really leave you hanging:
https://github.com/awslabs/aws-lambda-rust-runtime?tab=readme-ov-file#tracing-and-logging

**Reply by calavera** [reply]
If you want anything written in CloudWatch, you have to print it somehow. If you use the `tracing` crate, the messages will be written depending on the log level, and following the log level rules that `tracing` uses. The default log level is `INFO`, so it you use `tracing::error!`, messages will be written to CloudWatch.

**Reply by evbo** [reply]
Thanks @calavera that's actually gold nuggets for the docs. Currently [AWS docs](https://docs.aws.amazon.com/lambda/latest/dg/rust-logging.html#rust-logging-tracing) look like you need to use `instrument`, but based on what you're saying at a minimum we just need to set `LOG_LEVEL` env var to our liking and then do something like:

```rust
tracing_subscriber::fmt()
        .json()
        .with_max_level(tracing::Level::ERROR)
        .with_current_span(false)
        .with_ansi(false)
        .without_time()
        .with_target(false)
        .init();
```

With that any error should go to cloudwatch, right?
